src/Silhouette.js

```diff
@@ -67,12 +79,54 @@ class Silhouette {
         const imageData = ctx.getImageData(0, 0, width, height);
 
         this._data = new Uint8ClampedArray(imageData.data.length / 4);
+        this._colorData = imageData.data;
 
         for (let i = 0; i < imageData.data.length; i += 4) {
             this._data[i / 4] = imageData.data[i + 3];
         }
     }
 
+    colorAtNearest (vec) {
+        if (!this._colorData) {
+            return _default4b.slice();
+        }
+
+        return getColor4b(
+            this,
+            Math.floor(vec[0] * (this._width - 1)),
+            Math.floor(vec[1] * (this._height - 1))
+        );
+    }
+
+    colorAtLinear (vec) {
+        if (!this._colorData) {
+            return _default4b.slice();
+        }
+
+        const x = vec[0] * (this._width - 1);
+        const y = vec[1] * (this._height - 1);
+
+        const xDelta = x % 1;
+        const yDelta = y % 1;
+
+        const xFloor = Math.floor(x);
+        const yFloor = Math.floor(y);
+
+        const corners = [
+            getColor4b(this, xFloor, yFloor),
+            getColor4b(this, xFloor + 1, yFloor),
+            getColor4b(this, xFloor, yFloor + 1),
+            getColor4b(this, xFloor + 1, yFloor + 1)
+        ];
+
+        return Uint8ClampedArray.from({length: 4}, (_, i) => (
```

Comment:
Most VMs I believe will allocate a closure object here. Manually setting the indices with a helper function will be faster.